### PR TITLE
feat(cmf): add "component mode" instead of classic router mode to embed components in applications

### PR DESCRIPTION
**Pull request by talend-sromain**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

CMF at the moment only work to embed applications.

**What is the new behavior?**

CMF can embed component in case of bundled applications ( Semantic Dictionary )
If children is passed to CMF, CMF load children instead of use the router

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://109.talend.surge.sh)

File: packages/cmf/src/App.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ export default function App(props) {
 	return (
 		<Provider store={props.store}>
 			<RegistryProvider>
-				<UIRouter history={hist} />
+				{props.children !== undefined ? props.children : <UIRouter history={hist} />}
```
Comment (jsomsanith) [inline packages/cmf/src/App.js]: What about `{props.children || <UIRouter history={hist} />}` ?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://109.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://109.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://109.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://109.talend.surge.sh)

**Comment by talend-glorieux** [comment]
@talend-sromain Please fix lint issue.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://109.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://109.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://109.talend.surge.sh)

**Pull request merged**